### DISCUSSION TITLE
How to remove the default link argument ‘-Wl,—no-undefined’

### DISCUSSION THREAD

**Discussion by afud** [mesonbuild/meson#9777] [Q&A] [answered]
**Describe the bug**
Hi
my project be called by another project as a shared lib, and there is a func implemented in the caller project. The default argument ‘no-undefined’ will leading to a linking error, how can I remove it. 
Thanks.

**To Reproduce**
Please include your `meson.build` files, preferably as a minimal toy example showing the issue.
You may need to create simple source code files (don't include private/proprietary code).

**Expected behavior**

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
* what Python version are you using e.g. 3.8.0
* what `meson --version`
* what `ninja --version` if it's a Ninja build

**Comment by eli-schwartz** [comment, marked as answer]
Using one of:
- shared_module instead of shared_library, for dlopened plugins
- `override_options: ['b_lundef=false']`

**Reply by afud** [reply]
It’s amazing, thanks a lot.
